### ISSUE TITLE
knn_points return wrong distance value

### ISSUE THREAD

**Issue by Tendocat** [facebookresearch/pytorch3d#1946]
The knn_points dists returned values are different from the ones I'm computing myself given the knn.idx.
I except the same result than the one I am computing.

Here is the code (given two point clouds `vertices` and `other_vertices` of shape [batch, points, dim]):

```python
import torch
from pytorch3d.ops.knn import knn_points, knn_gather

def l2dist(a:torch.Tensor, b:torch.Tensor) -> torch.Tensor:
    diff = a - b
    return (diff*diff).sum(-1).sqrt()

knn = knn_points(vertices, other_vertices)
nn_vertices = knn_gather(other_vertices, knn.idx)[:,:,0]
distances = l2dist(nn_vertices, vertices)
print( knn.dists[:,:,0] - distances)
```
